canonical-web-and-design/maas-ui
# feat(ui): add description field for marking machine broken

## Done
A description can now be provided when marking a machine broken,
which now displays the text under status on the machine listing.

designs:
https://app.zeplin.io/project/5e6a623cde850113ba28d75b/screen/5e7d035047a0278e20c0aa72

![image](https://user-images.githubusercontent.com/130286/93556174-3d945f80-f9cc-11ea-9a4d-212b242e73ae.png)

![image](https://user-images.githubusercontent.com/130286/93556230-6288d280-f9cc-11ea-9fc3-6be61476cb09.png)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Select a few allocated machines, and select the mark broken action.
- Add a description and mark broken.
- You should see the mark broken description in the machine listing in the status column.

## Fixes
Fixes: canonical-web-and-design/MAAS-design#924

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
